src/metabase/query_processor/middleware/process_userland_query.clj

```diff
@@ -92,12 +92,12 @@
       ([acc]
        ;; We don't actually have a guarantee that it's from a card just because it's userland
        (when (integer? (:card_id execution-info))
-         (events/publish-event! :event/card-query {:card_id      (:card_id execution-info)
-                                                   :actor_id     (:executor_id execution-info)
-                                                   :cached       (:cached acc)
-                                                   :context      (:context execution-info)
-                                                   :ignore_cache (get-in execution-info
-                                                                         [:json_query :middleware :ignore-cached-results?])}))
+         (events/publish-event! :event/card-query {:user-id (:executor_id execution-info)
+                                                   :object  {:id           (:card_id execution-info)
+                                                             :cached       (:cached acc)
+                                                             :context      (:context execution-info)
+                                                             :ignore_cache (get-in execution-info
+                                                                                   [:json_query :middleware :ignore-cached-results?])}}))
```

Comment:
Why did you move this data under an `:object` key? It doesn't look like an object to me, just a bunch of data

Reply:
https://github.com/metabase/metabase/blob/7eb27d08d973ba987657d41316e8a48b6090a75b/src/metabase/events/view_log.clj#L81

this is the code that consumes this event and it's currently used to process `:event/dashboard-read`, `:event/table-read`, `:event/card-read` as well.

so I make it an object in order to not have to modify this defmethod.

another option is to create a new defmethod just to handle `:event/card-query`, is that better?

Reply:
Yes, I think creating a new defmethod to handle `:event/card-query` is a better option. We should avoid reusing names of keys inconsistently. The current handler for `:metabase.events.view-log/event` is effectively handling two different formats of data - one for actual views and one for queries.